src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -337,5 +338,34 @@ public String getStatusUpdates(int lastKnownSerial) {
       Log.i(TAG, "getStatusUpdates");
       return WebxdcActivity.this.dcContext.getWebxdcStatusUpdates(WebxdcActivity.this.dcAppMsg.getId(), lastKnownSerial    );
     }
+
+    @JavascriptInterface
+    public String sendToChat(String message) {
+      Log.i(TAG, "sendToChat");
+      try {
+        JSONObject jsonObject = new JSONObject(message);
+
+        String text = null;
+        byte[] data = null;
+        String name = null;
+        if (jsonObject.has("base64")) {
+            data = Base64.decode(jsonObject.getString("base64"), Base64.NO_WRAP | Base64.NO_PADDING);
+            name = jsonObject.getString("name");
+        }
+        if (jsonObject.has("text")) {
+            text = jsonObject.getString("text");
+        }
+
+        if (TextUtils.isEmpty(text) && TextUtils.isEmpty(name)) {
+            return "provided file is invalid, you need to set both name and base64 content";
+        }
+
+        DcHelper.share(WebxdcActivity.this, data, "application/octet-stream", name, text);
```

Comment:
i think, this line closess the  webxdc because of `launchMode="singleTask"` of some subsequent activities? if so, maybe add a little comment as that is not obvious.

```suggestion
        // this ends the WebxdcActivity as by `launchMode="singleTask"` an existing activity is restarted and removes all activities on top
        DcHelper.share(WebxdcActivity.this, data, "application/octet-stream", name, text);
```

i assume, btw, that the WebxdcActivity is not terminated my the chat-selection but on step further (did not test yet, however)

Reply:
that will result in the promise being rejected? that's probably fine then, even good for testing.

Reply:
i added a commit to https://github.com/webxdc/webxdc-test/pull/11 that prints the state of the promise - and, indeed, the promise is _resolved_ in that case (seems to happen implicitly when the function returns).

i think, this is okay: we're saying that the webxdc does not know anything about the sending state, it just drops a file, and that succeeded. but we need to update the docs, saying that the promise might be resolved.